libbeat/publisher/queue/spool/inbroker.go

```diff
@@ -214,30 +221,32 @@ func (b *inBroker) eventLoop() {
 
 	// notify ackLoop to stop only after eventLoop has finished (after last flush)
 	defer close(b.ackDone)
+	defer b.eventloopShutdown()
 
 	for {
 		ok := b.state(b)
 		if !ok {
 			break
 		}
 	}
+}
 
+func (b *inBroker) eventloopShutdown() {
 	// try to flush events/buffers on shutdown.
 	if b.bufferedEvents == 0 {
 		return
 	}
 
-	// try to append pending events
+	// Try to flush pending events.
+	w := b.writer
 	for len(b.pending) > 0 {
-		n, err := b.queue.Writer().Write(b.pending)
+		n, err := w.Write(b.pending)
```

Comment:
This error will get hidden, should we log it as debug?

Reply:
It's not logged on purpose. There can be a many reasons why this fails. E.g. IO error vs. spool is full. The later one is no error, but expected behavior. In future PRs we will check for the causes + handle/log them appropriately.